src/Executables/PrintMinimumGridSpacing/PrintMinimumGridSpacing.cpp

```diff
@@ -0,0 +1,154 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <boost/program_options.hpp>
+#include <iomanip>
+
+#include "Domain/InitialElementIds.hpp"
+#include "Domain/MinimumGridSpacing.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Parallel/Printf.hpp"
+#include "Options/ParseOptions.hpp"
+
+using frame = Frame::Inertial;
+
+class DimensionOption;
+struct Dimension {
+  using type = DimensionOption;
+  static constexpr OptionString help = {"help"};
+};
+
+class DimensionOption {
+ public:
+  struct Value {
+    using type = size_t;
+    static constexpr OptionString help = {
+        "Defines a number of dimensions (1, 2, or 3)"};
+  };
+  using options = tmpl::list<Value>;
+  static constexpr OptionString help = {
+      "Defines a number of dimensions (1, 2, or 3)"};
+  DimensionOption() = default;
+  DimensionOption(size_t value, const OptionContext& context)
+      : value_(std::move(value)) {
+    if (value_ != 1 and value_ != 2 and value_ != 3) {
+      PARSE_ERROR(context, "Dimemsion Value must be 1, 2, or 3 but is "
+                               << value_);
+    }
+  }
+  size_t value_{};
+};
+
+template <size_t Dim>
+void compute_and_print_minimum_grid_spacing(std::string input_file) {
```

Comment:
You can take the string by `const&`, i.e. `const std::string&` so that it doesn't need to be copied.

This function should be marked as `noexcept` since it doesn't need to throw for error handling purposes. That is, `... input_file) noexcept {`